### PR TITLE
LPS-185674 Create the model builder components

### PR DESCRIPTION
**Pull request by igor-franca**
Related [LPS-185674](https://liferay.atlassian.net/browse/LPS-185674)
Previous [PR](https://github.com/liferay-objects/liferay-portal/pull/2426) that did not start the tests

### CODE REVIEW

**Review by alinedoleron — APPROVED** [review]
LGTM!

**Comment by igor-franca** [comment]
ci:test:object

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ViewObjectDefinitionsDisplayContext.java [hunk 1]
```diff
@@ -124,6 +116,14 @@ public List<FDSActionDropdownItem> getFDSActionDropdownItems()
...
 	}
 
+	public String getModelBuilderURL() throws Exception {
+		return PortletURLBuilder.create(
+			getPortletURL()
+		).setMVCRenderCommandName(
+			"/object_definitions/edit_object_folder"
+		).buildString();
+	}
+
```
Comment (mateussgn) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ViewObjectDefinitionsDisplayContext.java:126]: Why did you rename but keep using [EditObjectFolder](https://github.com/liferay-objects/liferay-portal/blob/b4a34a493d15f69bd310256feb254303933b340a/modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx#L10)?

**Review by mateussgn — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ViewObjectDefinitionsDisplayContext.java (hunk 1 above)
Comment (igor-franca) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ViewObjectDefinitionsDisplayContext.java:126]: In this case I have the folders that will be edited within the model builder tool. So I tried to connect these two concepts.

I thought about renaming MVC Render and JSP from EditObjectFolder to ModelBuilder. But in BE we don't have the concept of ModelBuilder.

The name of the button that will take us to this page is called "View in Model Builder", I wanted to pass to the button a URL with a name referring to the ModelBuilder.

I really believe that we can connect these two concepts in a better way. Perhaps in the HashMapBuilder calls the modelBuilderURL key and calls the EditObjectFolderURL method. What do you think?

Comment (mateussgn) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ViewObjectDefinitionsDisplayContext.java:126]: I think we should maintain the pattern and name the method relating what it does.
In this case I suggest to keep the method name, and don't change it, since the method is returning the `edit_object_folder` URL.

Comment (igor-franca) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ViewObjectDefinitionsDisplayContext.java:126]: I came up with the name of this method in the previous PR and now I believe I can better name the MVC Render and JSP method to point to the ModelBuilder.

I'm thinking of changing rename the `EditObjectFolderMVCRenderCommand` to `ModelBuilderMVCRenderCommand`, the JSP `edit_object_folder.jsp` to `model_builder.jsp` and consequently the return of this method.

Comment (mateussgn) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ViewObjectDefinitionsDisplayContext.java:126]: Great suggestion, if you rename the others it will fit better.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_folders/view_model_builder.jsp [hunk 2]
```diff
@@ -10,6 +10,7 @@
 <%
 String backURL = ParamUtil.getString(request, "backURL", String.valueOf(renderResponse.createRenderURL()));
 ObjectDefinitionsDetailsDisplayContext objectDefinitionsDetailsDisplayContext = (ObjectDefinitionsDetailsDisplayContext)request.getAttribute(ObjectWebKeys.OBJECT_DEFINITIONS_DETAILS_DISPLAY_CONTEXT);
+ObjectDefinitionsRelationshipsDisplayContext objectDefinitionsRelationshipsDisplayContext = (ObjectDefinitionsRelationshipsDisplayContext)request.getAttribute(ObjectWebKeys.OBJECT_DEFINITIONS_RELATIONSHIP_DISPLAY_CONTEXT);
```
Comment (mateussgn) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/object_folders/view_model_builder.jsp]: I did not see this type of pattern before, using two different _display contexts_ at the same _jsp_.
But I believe it will not cause demages.

**Review by mateussgn — APPROVED** [review]
LGTM! 🎉

**Comment by igor-franca** [comment]
The next push will only resolve conflicts related to language properties:

**Comment by naraandrade** [comment]
ci:test:sf

**Comment by naraandrade** [comment]
ci:test:object

**Comment by naraandrade** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_folders/view_model_builder.jsp (hunk 2 above)
Comment (guilhermedcamacho) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/object_folders/view_model_builder.jsp]: Hey @mateussgn , I found a pattern using two display contexts in [asset-list-web/src/main/resources/META-INF/resources/init.jsp](https://github.com/liferay/liferay-portal/blob/0ab2d7fce2c261044150c108fdab2d3d4bfa4e90/modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/init.jsp#L102-L103).
I understood that he did this to reuse code from these two preexisting display contexts.
Another option would be to create something like ObjectFoldersViewModelBuilderDisplayContext class with these two inside.

**Review by guilhermedcamacho — APPROVED** [review]

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
ci:forward:force

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_folders/view_model_builder.jsp (hunk 2 above)
Comment (mateussgn) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/object_folders/view_model_builder.jsp]: Great information! Thanks for bring it up!
